docs/tutorials/my-first-custom-action.rst

```diff
@@ -0,0 +1,255 @@
+
+From the docs:
+
+.. pull-quote::
+    nodeSelector is the simplest recommended form of node selection constraint. nodeSelector is a field of PodSpec. It specifies a map of key-value pairs. For the pod to be eligible to run on a node, **the node must have each of the indicated key-value pairs as labels** (it can have additional labels as well). The most common usage is one key-value pair.
+
+So you comment out those lines, run kubectl apply again, and all is well.
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: nginx
+      labels:
+        env: test
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+        imagePullPolicy: IfNotPresent
+    #  nodeSelector:
+    #    spoiler: alert
+
+
+Automating the detection with a Robusta Playbook
+----
+What we need to do?
+----
+
+.. note::
+    Make sure to clean up the pod from the last section by running ``kubectl delete pod nginx``
+
+A playbook consists of two things:
+
+- Trigger - We’re gonna use a built in trigger
+- Action - We’re gonna write our own action!
+
+
+Finding the correct trigger
+----
+So what is the correct trigger for the job?
+We can think of two triggers that may fit:
+
+- Creation of a new pod (because we create a new pod, ‘ngnix’)
+- A Kubernetes Event is fired (because we ran kubectl get event to find out the scheduling error)
+
+Let’s look at the Trigger section about :ref:`Kubernetes (API Server)`, and try to find out triggers for both.
+Go ahead and try to find them!
+
+Okay! We find ``on_pod_create`` and ``on_event_create``
+
+We’ll use ``on_event_create``, because in this case ``on_pod_create`` is not even called because the pod is not created.
+
+Writing the action
+----
+
+Now we need to write code that checks this event and reports it. To find out the correct event class that matches our trigger ``on_event_create``. please take a look at :ref:`Event Hierarchy`.
+
+Okay! We find out it’s ``EventEvent``!
+
+So we need to get the information, check for the scenario, and then report it (for more information about reporting it see :ref:`Findings API`)
+
+Let’s name our action ``report_scheduling_failure``, and write everything in a python file:
+
+.. code-block:: python
+
+    from robusta.api import *
+
+    @action
+    def report_scheduling_failure(event: EventEvent):
+        actual_event = event.get_event()
+
+        print(f"This print will be shown in the robusta logs={actual_event}")
+
+        if actual_event.type.casefold() == f'Warning'.casefold() and \
+            actual_event.reason.casefold() == f'FailedScheduling'.casefold() and \
+            actual_event.involvedObject.kind.casefold() == f'Pod'.casefold():
+            _report_failed_scheduling(event, actual_event.involvedObject.name, actual_event.message)
+
+    def _report_failed_scheduling(event: EventEvent, pod_name: str, message: str):
+        # this is how you send data to slack or other destinations
+        event.add_enrichment([
```

Comment:
I think we need to create a finding here. 
Otherwise, the default finding is used, which has some generic fields

Reply:
what generic fields bug you? I think if we can keep the code short and simple it is best

Reply:
The finding title would be: "Robusta notification"
The aggregation key (which is used for grouping in the timeline) would be: "Generic finding key"

This would make the example a little longer though:
`   
    finding = Finding(
        title=f"Pod {pod_name} Failed Scheduling",
        source=FindingSource.KUBERNETES_API_SERVER,
        aggregation_key="Pod Failed Scheduling",
    )
    finding.add_enrichment([...])
    event.add_finding(finding)
`

Reply:
I think it makes it more complicated, so I won't use it.

Reply:
Your call obviously.
But I think calling only event.add_enrichment() isn't the correct use of our API.
This approach should be used only when chaining few playbooks using the same finding, like in alerts enrichers.
There, the created finding has meaning.

In general, if one needs to create a finding, he needs to create both the finding and the enrichment.
@aantn your thoughts about this?

Reply:
@arikalon1 you're correct (and we should document it somewhere) but for the sake of the tutorial I think either is fine.